### ISSUE TITLE
how can I use this project to train a model for Chinese ?

### ISSUE THREAD

**Issue by hesilong** [karpathy/nanoGPT#109]
Hi, 
   I'm an engineer from China. This is a wonderful project. I want to know whether I can use this project to train a model for Chinese or not, thanks!

**Comment by codetiger** [comment]
Yes you can. I am using this project for other languages and it works very well. You might have to tune the encoder and decoder a little bit to improve for other languages.

**Comment by hesilong** [comment]
> Yes you can. I am using this project for other languages and it works very well. You might have to tune the encoder and decoder a little bit to improve for other languages.

thanks for your reply! I will have a try!

**Comment by venzen** [comment]
@codetiger When you say "tune the encoder/decoder a little bit", can you give a brief explanation or example?

**Comment by codetiger** [comment]
@venzen In every language there are sequence of characters that are used often. Ex: "tion", "ble", "ve", and "es". These sequences can be encoded using a separate id to achieve better results. In my local language there are lot of such combinations. Having these in the encoder helps the model to infer grammar much better.

**Comment by vgoklani** [comment]
@codetiger are you using the OpenAI encoder or a different model (say huggingface's tokenizer)?

**Comment by codetiger** [comment]
I tried various encoders and finally settled down with ByteLevelBPETokenizer

**Comment by JeckerWen** [comment]
你好，训练一个中文GPT，你解决的么？ 可不可以提供一个技术方案
